### PR TITLE
feat: advanced search #799 functionality

### PR DESCRIPTION
**Pull request by tomaszandel**
Figma: https://www.figma.com/file/thf78Tiwv9qNGuorwLG1ti/Search?type=design&node-id=691%3A1242&mode=design&t=Qo6m23L1pfqbe5Ww-1

### CODE REVIEW

**Comment by Marcelinna** [comment]
@agpul @aleksandrahojszyk - Want to add it? do we have any text for this?
image

**Comment by Marcelinna** [comment]
Why do I have a difference between keywords in advanced search and filter

Advance Search - 7 result
image

Filter - 5 result 
image

**Comment by Marcelinna** [comment]
@aleksandrahojszyk @agpul @tomaszandel 
Should we be able to search by keywords in a collection that doesn't have it ?
image

**Comment by Marcelinna** [comment]
We don't have keyword searches for services, datasources

image
image

**Comment by Marcelinna** [comment]
we don't have to highlight for characters similar to المغرب
image

**Comment by agpul** [comment]
> @agpul @aleksandrahojszyk - Want to add it? do we have any text for this? image

Yes, we need to add this @tomaszandel, text for this tooltip should be provided by @eldenkat

**Comment by agpul** [comment]
> @aleksandrahojszyk @agpul @tomaszandel Should we be able to search by keywords in a collection that doesn't have it ? user always get no data Guidelines, Bundle - don't have keywords image

We should remove search by keywords for collections that do not have keywords.

**Comment by tomaszandel** [comment]
> Why do I have a difference between keywords in advanced search and filter ?
> 
> Advance Search - 7 result image
> 
> Filter - 5 result image

I think its ok, as search is in fuzzy mode

> Why do I have a difference between keywords in advanced search and filter ?
> 
> Advance Search - 7 result image
> 
> Filter - 5 result image

This is ok. keywords are also partially matched, and filters are not: hierarchical Classification vs Classification

**Comment by eldenkat** [comment]
the slider doesn't seem to work:

1. Software -> keywords - "open data" or "HPC" - 247 search results (ANY)

Zrzut ekranu 2023-08-31 142553

2. Software -> keywords - "open data" and "HPC" - 247 search results (ALL)

Zrzut ekranu 2023-08-31 142835

3. But...in second search there are records containing only one (not all) chosen keyword

Zrzut ekranu 2023-08-31 145023

**Comment by eldenkat** [comment]
Still "no data" in search by keywords in **datasources** and **services**

**Comment by eldenkat** [comment]
Does search ignores capital letters?

1.  Go to software -> author:gleeson + author: Sinha
2. See 37 items:

Zrzut ekranu 2023-09-01 080955

3. Add exact: neuroml
4. See 34 items with "neuroml" highlighted, but there are also 3 items with "neuroML" that are missed (see screenshot above):

Zrzut ekranu 2023-09-01 081557

**Comment by eldenkat** [comment]
In first and second line of operators "all" is shown by deafault
In third line of operators "any" is shown by deafault
In "none of" operator there's no slider all/any

Zrzut ekranu 2023-09-01 085539

**Comment by tomaszandel** [comment]
> 
Yup exact is exact ;)

**Comment by tomaszandel** [comment]
> 

yup this is because im not clearing previous values, it could also be that it is set in the browser path, to be fixed

**Comment by eldenkat** [comment]
- [ ] Since there is no "**author**" for the **datasource**, and we already have working "**organization**" filter - remove the "author" operator from advanced search.
- [ ] Since there is no "**author**" for the **services**, and we already have working "**organization**" filter - remove the "author" operator from advanced search.
- [ ]  Since there is no "**author**" for the **IGs**, and we already have working "**provider**" filter - remove the "author" operator from advanced search.

**Comment by eldenkat** [comment]
Names do not add to rows in the order they are typed (not necessarily a problem):

Zrzut ekranu 2023-09-01 134353

**Comment by eldenkat** [comment]
With the "exact" operator, records are searched both with and without a special character in the typed query. But highlights are only in results with special char. 

Zrzut ekranu 2023-09-01 141244

**Comment by eldenkat** [comment]
To sum up:
**1. Slider comment -done (responsive)**
2. No data in keywords operator (datasources and services) - still not working
3. Exact match still separates uppercase and lowercase letters in the same phrase (eg. when search neuroml it doesnt highlight neuroML)
4. In every operator query is shown by deafault as "all" - done
5. None of is still muted as it comes to all/any slider - dunno, maybe it works as it sould (@tomaszandel )
**6. Remove author from Datatsources, Services and IGs - done**
7. With the "exact" operator, records are searched both with and without a special character in the typed query. But highlights are only in results with special char. - not done

**Comment by aleksandrahojszyk** [comment]
AS poprawki
1. Jeśli jest więcej niż jedna linia “chipsów” to przesuńmy margines drugiej i kolejnych do linii
2. All / Any -> And / Or
3. Clear All filters -> Clear Advanced search i na biało
4. tekst do tootlip: "Choose this for precise results. Only content with the exact phrase you entered will show up. No changes or variations."
5. wyrownać elementy exact match
6. czy zamiast iksa w chipsach są 'X' ? zmienić na ikonę 
7. x

**Comment by tomaszandel** [comment]
> AS poprawki
> 
> 1. Jeśli jest więcej niż jedna linia “chipsów” to przesuńmy margines drugiej i kolejnych do linii
> 2. All / Any -> And / Or
> 3. Clear All filters -> Clear Advanced search i na biało
> 4. tekst do tootlip: "Choose this for precise results. Only content with the exact phrase you entered will show up. No changes or variations."
> 5. wyrownać elementy exact match
> 6. czy zamiast iksa w chipsach są 'X' ? zmienić na ikonę
> 7. x
1. ok
2. nie intuicyjne imo And i OR albo All ANy, mieszanie Any Or nie ma imo sensu, ale jak chcecie
3. jw + brakuje bialej ikony w assretach jest niebieska i w figmie
4. ok
5. ok
6. sa ikony ale powiekszone, moge zmniejszyc jak cos

**Comment by aleksandrahojszyk** [comment]
@tomaszandel tak, chodziło mi o całkiem przejście z All / Any na And / Or a nie mieszanie – rekomendacja Ani była żeby zostać przy Or / And ze względu na to że jest to standard przyjęty

**Comment by aleksandrahojszyk** [comment]
Screenshot 2023-09-11 at 14 18 45
Change to And / Or a nie All / Or

**Comment by eldenkat** [comment]
@aleksandrahojszyk nie mieliśmy zmienić koloru AND/OR I Clear advanced search?

**Comment by aleksandrahojszyk** [comment]
Screenshot 2023-09-13 at 09 40 28
zmienić kolor tekstu w switchu na biały i clear advanced search też na biało

**Comment by aleksandrahojszyk** [comment]
Screenshot 2023-09-13 at 09 43 57 
oraz wyrównać wielkości elementów

**Comment by Marcelinna** [comment]
Author for bundle should be not available
image

**Comment by Marcelinna** [comment]
The author for provider should be not available
Providers have keywords but we don't show them now in UI and I see that the search returns no data when I try to search by keywords, so we should hide keywords for this moment
image
image
image

**Comment by Marcelinna** [comment]
Step to reproduce
1. Type "Comparaison" in search bar for Publications - I see 405 result
image
2. Change collection to Data - 69 result
image
3. Change collection to Provider - 520 result
image
4. Click loupe icon - 0 result
image

In step 3 I see problem, result should be 0

**Review by Michal-Kolomanski — COMMENTED** [review]
@agpul @tomaszandel 
1. I do not see an advanced search at all (maybe I dont know where to look). Is it successfully built?
image
2. Its my personal opinion but I find it weird that when I type text in the search bar - the portal acts as results are continuously calculated based on what I am typing (enter not needed, results disappear and appear) but in reality, results are only calculated after the enter action. Confusing to me. 
3. Another personal opinion: I am not sure if selecting/deselecting "Exact match" should immediately trigger a searching action. I would rather first prepare my search query and then press enter or click on the loupe when ready.
4.  Please rebase this branch on the latest changes. There are many conflicts. Also, I assume @Marcelinna finds it difficult to test it robustly due to that.

**Comment by agpul** [comment]
> @agpul @tomaszandel
> 
> 1. Its my personal opinion but I find it weird that when I type text in the search bar - the portal acts as results are continuously calculated based on what I am typing (enter not needed, results disappear and appear) but in reality, results are only calculated after the enter action. Confusing to me.
> 2. Another personal opinion: I am not sure if selecting/deselecting "Exact match" should immediately trigger a searching action. I would rather first prepare my search query and then press enter or click on the loupe when ready.
> 3. Please rebase this branch on the latest changes. There are many conflicts. Also, I assume @Marcelinna finds it difficult to test it robustly due to that.

1. this is not the scope of the issue, you are talking about autocomplete
2. probably not possible right now
3. it was rebased, some new changes are on beta

**Review by michal-szostak — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "[python]": {
+        "editor.defaultFormatter": "ms-python.black-formatter"
+    },
+    "python.formatting.provider": "none"
+}
```
[the remark this answers is not in the document]
Comment (tomaszandel) [reply .vscode/settings.json:6]: yeah.. i added .vscode to ignore now, strange it wasnt there

**Pull request merged**